SalesforceCommerceCloud/pwa-kit
# [Retail App] When a registered shopper logs in the items in the guest cart should move to the registered cart

https://github.com/SalesforceCommerceCloud/pwa-kit/issues/319

# Description

When a registered shopper logs in the items in the guest cart should move to the registered cart. Created an extension to `OcapiShoppersBasket` class to include SCAPI ShopperBasket function to `mergeBasket`.

# Types of Changes


> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Added `mergeBasket` function to `useBasket` hook
- Added `ShopperBaskets` class an an extension to `OcapiShopperBaskets` and implemented SCAPI `mergeBasket` function
- Added guest usid to login call


# How to Test-Drive This PR

- Log out of the retail app
- Add item to cart
- Login into the retail app
- Make sure the cart items are merged after successful login

# Checklists

## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
